common/src/api/external/mod.rs

```diff
@@ -296,6 +296,85 @@ impl Name {
     }
 }
 
+/**
+ * Name for a built-in role
+ */
+#[derive(
+    Clone,
+    Debug,
+    DeserializeFromStr,
+    Display,
+    Eq,
+    FromStr,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    SerializeDisplay,
+)]
+#[display("{resource_type}.{role_name}")]
+pub struct RoleName {
+    #[from_str(regex = "[a-z_]+")]
```

Comment:
Should we use - instead of _ to be consistent with name strings?

Reply:
Done in 38fccbff.  I changed this in both the resource type and the role name.